### PR TITLE
keycloak_client_rolemapping.py: add support for subgroups

### PR DESCRIPTION
**Pull request by mikhail-putilov**
##### SUMMARY
There was a missing `parents` parameter which prevented my ansible scripts from mapping roles to subgroups

The fix was easy, as everything was already implemented and only missing `parents` parameter needed to be added.

##### ISSUE TYPE
- Feature Pull Request

##### COMPONENT NAME
keycloak_client_rolemapping

##### ADDITIONAL INFORMATION
I needed to map a role to a subgroup but I could not because there was a missing parameter `parents` in the python script. Adding it solved the issue.

### CODE REVIEW

File: plugins/modules/keycloak_client_rolemapping.py [hunk 1]
```diff
@@ -144,6 +171,24 @@
         id: role_id2
   delegate_to: localhost
 
+- name: Map a client role to a subgroup, authentication with token
```
Comment (mikhail-putilov) [inline plugins/modules/keycloak_client_rolemapping.py:174]: This is a copy of YAML above with one additional `parents` parameter and slightly changed `name`

File: plugins/modules/keycloak_client_rolemapping.py [hunk 2]
```diff
@@ -63,6 +63,33 @@
             - Name of the group to be mapped.
             - This parameter is required (can be replaced by gid for less API call).
 
+    parents:
```
Comment (mikhail-putilov) [inline plugins/modules/keycloak_client_rolemapping.py:66]: This is a copy of a documentation from https://github.com/ansible-collections/community.general/blob/779a5d99f5500bf3b228783b44c4cc32c1c6fbd2/plugins/modules/keycloak_group.py#L79 slightly adjusted

File: changelogs/fragments/6687-support-subgroups-for-keycloak-client-rolemapping.yml [hunk 3]
```diff
@@ -0,0 +1,2 @@
+minor_changes:
+  - keycloak_client_rolemapping - adds support for subgroups with additional parameter ``parents``
```
Comment (felixfontein) [inline changelogs/fragments/6687-support-subgroups-for-keycloak-client-rolemapping.yml:2]:
```suggestion
  - keycloak_client_rolemapping - adds support for subgroups with additional parameter ``parents`` (https://github.com/ansible-collections/community.general/pull/6687).
```

File: plugins/modules/keycloak_client_rolemapping.py [hunk 4]
```diff
@@ -63,6 +63,33 @@
...
+        version_added: "6.4.0"
```
Comment (felixfontein) [inline plugins/modules/keycloak_client_rolemapping.py:67]:
```suggestion
        version_added: "7.1.0"
```

File: plugins/modules/keycloak_client_rolemapping.py [hunk 5]
```diff
@@ -63,6 +63,33 @@
...
+            - List of parent groups for the group to handle sorted top to bottom.
+            - >-
+              Set this if your group is a subgroup and your do not provide GID
+        elements: dict
+        suboptions:
+          id:
+            type: str
+            description:
+              - Identify parent by ID.
+              - Needs less API calls than using I(name).
```
Comment (felixfontein) [inline plugins/modules/keycloak_client_rolemapping.py:79]:
```suggestion
              - Needs less API calls than using O(parents[].name).
```

File: plugins/modules/keycloak_client_rolemapping.py [hunk 6]
```diff
@@ -63,6 +63,33 @@
...
+              - A deep parent chain can be started at any point when first given parent is given as ID.
+              - Note that in principle both ID and name can be specified at the same time
+                but current implementation only always use just one of them, with ID
+                being preferred.
+          name:
+            type: str
+            description:
+              - Identify parent by name.
+              - Needs more internal API calls than using I(id) to map names to ID's under the hood.
```
Comment (felixfontein) [inline plugins/modules/keycloak_client_rolemapping.py:88]:
```suggestion
              - Needs more internal API calls than using O(parents[].id) to map names to ID's under the hood.
```

File: plugins/modules/keycloak_client_rolemapping.py [hunk 7]
```diff
@@ -63,6 +63,33 @@
             - Name of the group to be mapped.
             - This parameter is required (can be replaced by gid for less API call).
 
+    parents:
+        version_added: "6.4.0"
+        type: list
+        description:
+            - List of parent groups for the group to handle sorted top to bottom.
+            - >-
+              Set this if your group is a subgroup and your do not provide GID
```
Comment (felixfontein) [inline plugins/modules/keycloak_client_rolemapping.py:72]:
```suggestion
              Set this if your group is a subgroup and your do not provide GID.
```

**Review by felixfontein — COMMENTED** [review]
Thanks for your contribution! I've added some first comments.

**Review by felixfontein — APPROVED** [review]
Looks good to me (without really knowing keycloak). If nobody objects, I'll merge on Sunday evening.

**Comment by felixfontein** [comment]
@mikhail-putilov thanks for your contribution!

**Comment by mikhail-putilov** [comment]
I have to thank you for reviewing it so fast 👍

**Pull request merged**